packages/spec-core/src/scenarios-resolver.ts

```diff
@@ -34,7 +34,9 @@ export async function findScenarioSpecFiles(scenariosPath: string): Promise<Spec
     `${normalizedScenarioPath}/**/main.tsp`,
   ];
   logger.debug(`Looking for scenarios in ${pattern}`);
-  const fullScenarios = await findFilesFromPattern(pattern);
+  const fullScenarios = (await findFilesFromPattern(pattern)).map((scenario) =>
+    path.normalize(scenario),
```

Comment:
why do you need to normalize here? and what does it do? `\` for windows or `/` all the time. We use `normalizePath` already in places to normalize to unix `/`

Reply:
**Reason for including this code change**

If I do not include this mapping, the `fullScenarios` array element will be (after line 36) such as:

`C:/Users/sarajama/Projects/typespec-azure/packages/azure-http-specs/specs/client/structure/client-operation-group/main.tsp`. 

There is no problem in this line. But, in the following line 47, the condition:

`scenarioSet.has(join(dirname(scenario), "client.tsp"))` 

will always return false even though there is a `client.tsp` file in that directory. Why? If you look at the statement `join(dirname(scenario), "client.tsp")` it will return the value: 

`C:\\Users\\sarajama\\Projects\\typespec-azure\\packages\\azure-http-specs\\specs\\client\\structure\\client-operation-group\\client.tsp`. 

Whereas the `scenarioSet` has the value `C:/Users/sarajama/Projects/typespec-azure/packages/azure-http-specs/specs/client/structure/client-operation-group/main.tsp`. So, it will return false. As a result, the script `pnpm validate-scenarios` will fail. 

**How is this scenario handled in `cadl-ranch` repository?**

If the above change is required in this repository, how is the scenario successful in `cadl-ranch` repository? In the `cadl-ranch` repository, the `fullScenarios` array element will be such as:

`C:\\Users\\sarajama\\Projects\\cadl-ranch\\packages\\cadl-ranch-specs\\http\\client\\structure\\client-operation-group\\main.tsp`

So, this actually succeeds in the condition `scenarioSet.has(join(dirname(scenario), "client.tsp"))` and returns true. 

**Why is the difference?**

So, what exactly changed? Why is there a difference between the same code between `cadl-ranch` and `typespec` repositories? The difference is in the library used `glob` vs `globby`. In the `cadl-ranch` repository the code is as follows:

```
export async function findFilesFromPattern(pattern: string | string[]): Promise<string[]> {
  return await glob(pattern);
}
```

Whereas in the `typespec` repository, it is:

```
export async function findFilesFromPattern(pattern: string | string[]): Promise<string[]> {
  return await globby(pattern);
}
```

If I change it to `glob` dependency, then this code change is not required. It will work fine similar to `cadl-ranch` repository. But, it will need other changes such as `skipLibCheck: true` as described in https://github.com/isaacs/node-lru-cache/issues/354.

Reply:
Per offline discussion, changed the logic to use normalizePath after join.